platform-core-module/src/main/java/com/softicar/platform/core/module/program/QueuedProgramExecutionDaemon.java

```diff
@@ -102,6 +117,40 @@ private void terminate(Collection<IRunnableThread<ProgramExecutionRunnable>> run
 		}
 	}
 
+	private void checkMaximumRuntime(AGProgram program, AGProgramExecution currentExecution) {
+
+		var scheduledExecution = AGScheduledProgramExecution.getByAGUuid(program.getProgramUuid());
+		if (scheduledExecution != null) {
+			checkIfExceeded(program, currentExecution, scheduledExecution);
+		}
+	}
+
+	private void checkIfExceeded(AGProgram program, AGProgramExecution currentExecution, AGScheduledProgramExecution scheduledExecution) {
```

Comment:
Suggesting the following sequence of refactorings:
1. I think this method should return a `boolean`: `true` if runtime is exceeded, `false` otherwise.
2. If true is returned, we do this in `checkMaximumRuntime` (and no longer in `createEventAndTerminateIfConfigured`):
```
if (scheduledExecution.isAutoKill()) {
	terminateProgram(program);
}
```
3. Rename `createEventAndTerminateIfConfigured` to `createRuntimeExceededEvent`
4. Rename `checkMaximumRuntime` to `handleMaximumRuntime`.

This way, we'd need a nice separation of concerns (event creation vs. program termination), and no more confusing method names.